ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Connection.scala

```diff
@@ -51,7 +51,21 @@ private[h2] class H2Connection[F[_]](
     logger: Logger[F],
 )(implicit F: Temporal[F]) {
 
-  private[this] def addrStr = address.fold(_.toString, _.toString)
+  private[this] var _addrStr: String = null
+
+  private[this] def addrStr = {
+    if (_addrStr eq null)
+      _addrStr = address.fold(_.toString, _.toString)
+    _addrStr
+  }
+
+  private[this] var _connectionTerminated: String = null
+
+  private def connectionTerminated: String = {
+    if (_connectionTerminated eq null)
+      _connectionTerminated = s"Connection $addrStr readLoop Terminated"
+    _connectionTerminated
+  }
```

Comment:
Sorry to complain, but I don't like this at all 😢

firstly, it's a lot of code just to save some ops for non-happy path error/warning logging.

secondly, this is a bespoke encoding of `@unsafe lazy val`. if you insist to do that, then can we at least make a re-usable utility class instead of inlining all of this logic here? Or, we could just use `lazy val` or just use `val` or just leave it alone 😬